dbt/adapters/bigquery/relation_configs/materialized_view.py

```diff
@@ -0,0 +1,192 @@
+            "partition_by": model_node.config.extra.get("partition_by"),
+            "partition_expiration_date": model_node.config.extra.get("partition_expiration_date"),
+            "refresh_interval_minutes": model_node.config.extra.get("refresh_interval_minutes"),
+            "hours_to_expiration": model_node.config.extra.get("hours_to_expiration"),
+            "max_staleness": model_node.config.extra.get("max_staleness"),
+            "kms_key_name": model_node.config.extra.get("kms_key_name"),
+            "friendly_name": model_node.config.extra.get("friendly_name"),
+            "description": model_node.config.extra.get("description"),
+            "labels": model_node.config.extra.get("labels"),
+        }
+
+        raw_autorefresh_value = model_node.config.extra.get("enable_refresh")
+        auto_refresh_value = bool_setting(raw_autorefresh_value)
+        config_dict["enable_refresh"] = auto_refresh_value
+
+        return config_dict
+
+    @classmethod
+    def parse_relation_results(cls, relation_results: RelationResults) -> dict:
+        materialized_view_config = relation_results.get("materialized_view")
+        if isinstance(materialized_view_config, agate.Table):
+            materialized_view = cls._get_first_row(materialized_view_config)
+        else:
+            raise DbtRuntimeError("Unsupported type returned ex. None")
+
+        config_dict = {
+            "materialized_view_name": materialized_view.get("materialized_view_name"),
+            "schema_name": materialized_view.get("schema"),
+            "database_name": materialized_view.get("database"),
+            "cluster_by": materialized_view.get("cluster_by"),
+            "partition_by": materialized_view.get("partition_by"),
+            "partition_expiration_date": materialized_view.get("partition_expiration_date"),
+            "enable_refresh": materialized_view.get("enabled_refresh"),
+            "refresh_interval_minutes": materialized_view.get("refresh_interval_minutes"),
+            "hours_to_expiration": materialized_view.get("hours_to_expiration"),
+            "max_staleness": materialized_view.get("max_staleness"),
+            "kms_key_name": materialized_view.get("kms_key_name"),
+            "friendly_name": materialized_view.get("friendly_name"),
+            "description": materialized_view.get("description"),
+            "labels": materialized_view.get("labels"),
+        }
+
+        return config_dict
+
+
+@dataclass(frozen=True, eq=True, unsafe_hash=True)
+class BigQueryAutoRefreshConfigChange(RelationConfigChange):
+    context: Optional[bool] = None
+
+    @property
+    def requires_full_refresh(self) -> bool:
+        return False
+
+
+@dataclass(frozen=True, eq=True, unsafe_hash=True)
+class BigQueryPartitionConfigChange(RelationConfigChange):
+    context: Optional[FrozenSet[Dict[str, Any]]] = None
+
+    @property
+    def requires_full_refresh(self) -> bool:
+        return True
```

Comment:
@McKnight-42 look at diskey on dbt-redshift to see how to create combined objects for here